src/store/slices/playlists/slice.ts

```diff
@@ -60,8 +60,7 @@ const verifyUrl = createAsyncThunk<
             if (!videoDurationIsOk) {
                 return thunkApi.rejectWithValue("video_too_long")
             }
-
-            thunkApi.dispatch(addSong({youtubeId, title: videoDetails.title}))
+            thunkApi.dispatch(checkIfSongExists({youtubeId, title: videoDetails.title}))
```

Comment:
i dont think this is a great structure. why is the checkIfSongExists add a new song? why do we need to check if there is a user/playlist when the method named as checkIfSongExists?  (also these are checked in the add song thunk as well). this thunk should have only one job, and that is to call the db if there is a song and return the result

Reply:
Checking the currentuser is a mistake, I forgot to delete it. Now deleted. 
Checking the currentplaylist is needed though, as I need to playlistid from there. Otherwise there is a type error.